azerothcore/azerothcore-wotlk
# fix(Core/PlayerQuest): Get base skill value for quest prereqs.

## Changes Proposed:
- Replaces a `GetSkillValue` with `GetBaseSkillValue` when checking quest requisites for a player.

## Issues Addressed:

- Closes unreported issue.

## SOURCE:

Wrath Classic PTR, same scenario as the test performed/suggested test.

## Tests Performed:

- Rudimentary test with [one quest](https://www.wowhead.com/wotlk/quest=6607), one [skill increasing consumable](https://www.wowhead.com/wotlk/item=6533), and one [skill increasing equippable item](https://www.wowhead.com/wotlk/item=12225).

## How to Test the Changes:

1. `.go xyz -4139.7 -4058.2 2.3 1.9 1`
2. `.setskill 356 125 300`
3. `.additem 6533 5`
4. `.additem 12225`
5. Equip the fishing pole and use the lure, observe that the quest (with a 225 skill prereq for fishing) is not available.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
